src/cryptoadvance/specter/server_endpoints/settings.py

```diff
@@ -385,13 +408,25 @@ def auth():
             app.specter.check()
         elif action == "adduser":
             if current_user.is_admin:
-                new_otp = random.randint(100000, 999999)
+                new_otp = secrets.token_urlsafe(16)
+                now = time.time()
+                timeout = int(registration_link_timeout)
+                timeout = 0 if timeout < 0 else timeout
+                if timeout > 0:
+                    expiry = now + timeout * 60 * 60
+                    if timeout > 1:
+                        expiry_desc = " (expires in {} hours)".format(timeout)
+                    else:
+                        expiry_desc = " (expires in 1 hour)"
+                else:
+                    expiry = 0
+                    expiry_desc = ""
                 app.specter.add_new_user_otp(
-                    {"otp": new_otp, "created_at": time.time()}
+                    {"otp": new_otp, "created_at": now, "expiry": now}
```

Comment:
```suggestion
                    {"otp": new_otp, "created_at": now, "expiry": expiry}
```

Looks like here it should be expiry instead of now

Reply:
done